updatecli/weekly.d/contributing.yaml

```diff
@@ -0,0 +1,82 @@
+name: Update Technical Requirements in CONTRIBUTING.md
+
+sources:
+  getMavenLabel:
+    kind: file
+    name: Get Maven label from Jenkinsfile
+    spec:
+      file: Jenkinsfile
+      marker: 'label '''
+      pattern: '(maven-[0-9]+)'
+
+  getPackerImageVersion:
+    kind: yaml
+    name: Get Packer image version from jenkins-infra
+    spec:
+      file: hieradata/common.yaml
+      key: $.profile::jenkinscontroller::jcasc.agent_images.azure_vms_gallery_image.version
+
+  getToolVersions:
+    kind: yaml
+    name: Get tool versions from packer-images
+    dependson:
+      - getPackerImageVersion
+    spec:
+      file: https://raw.githubusercontent.com/jenkins-infra/packer-images/{{ source "getPackerImageVersion" }}/provisioning/tools-versions.yml
+
+conditions:
+  checkMavenAvailability:
+    kind: shell
+    disablesourceinput: true
+    spec:
+      command: curl --connect-timeout 5 --location --head --fail --silent --show-error https://archive.apache.org/dist/maven/maven-3/{{ source "getToolVersions" "maven.version" }}/binaries/apache-maven-{{ source "getToolVersions" "maven.version" }}-bin.tar.gz
+
+  checkJDKAvailability:
+    kind: shell
+    disablesourceinput: true
+    spec:
+      command: curl --connect-timeout 5 --location --head --fail --silent --show-error https://adoptium.net/temurin/releases/v{{ source "getToolVersions" "jdk.version" }}
+
+  checkPackerImageExists:
+    kind: http
+    name: Verify packer-images tools-versions.yml exists
+    spec:
+      url: https://raw.githubusercontent.com/jenkins-infra/packer-images/{{ source "getPackerImageVersion" }}/provisioning/tools-versions.yml
+
+targets:
+  updateTechnicalRequirements:
+    name: Update Technical Requirements section in CONTRIBUTING.md
+    kind: file
+    spec:
+      file: CONTRIBUTING.md
+      transformers:
+        - replaceRegex:
+            pattern: '### Technical Requirements[\s\S]*?(?=### Add a)'
+            with: |
+              ### Technical Requirements
+
+              In order to work with that repository you will need:
+
+              - JDK {{ source "getToolVersions" "jdk.version" }}: Check [Temurin releases](https://adoptium.net/temurin/releases/v{{ source "getToolVersions" "jdk.version" }})
+              - Maven {{ source "getToolVersions" "maven.version" }}: Check [Maven releases](https://maven.apache.org/download.cgi)
+              - Groovy: https://groovy-lang.org/install.html
+
+              The CI environment uses the `{{ source "getMavenLabel" }}` container image which is defined in the [jenkins-infra repository](https://github.com/jenkins-infra/jenkins-infra/blob/production/hieradata/common.yaml) and built using the [packer-images](https://github.com/jenkins-infra/packer-images) repository.
```

Comment:
This is rather "fragile" addition in the sense that the values are subject to change quote often and ill be quickly out of date: what if the Label changes or if we change the hieradata directory ?

=> I would rephrase to make the reader aware of the pipeline agent's label and link to https://github.com/jenkins-infra/documentation/blob/main/ci.adoc#tools-labels which is less subject to change, OR remove this block at all